Rubberduck.Core/AppMenu.cs

```diff
@@ -33,10 +33,34 @@ public AppMenu(IEnumerable<IParentMenuItem> menus, IParseCoordinator parser, ISe
 
         public void Initialize()
         {
-            _stateBar.Initialize();
+            InitializeRubberduckCommandBar();
+            InitializeRubberduckMenus();
+        }
+
+        private void InitializeRubberduckCommandBar()
+        {
+            try
+            {
+                _stateBar.Initialize();
+            }
+            catch (Exception exception)
+            {
+                _logger.Error(exception);
```

Comment:
Because we are now logging the exception, the side effect is that the code will continue onto to the next block instead of bailing out completely. While I think it's OK for later steps, I'm not sure if this is appropriate to keep trying if the initialization has failed --- do we want to `throw` in this case?

Reply:
Without assuming the cause or type of the exception, I think it's best to attempt to initialize as much as we can - could be just one command throwing for whatever buggy reason, let its command not be on the menu, but let's still have a menu if we can get one, no?

I'm over-thinking this, am I?

Reply:
Some thoughts... `_stateBar.Initialize()` does all the plumbing that, agreed, could throw stuff that's is probably better off rethrown. Problem is that it also initializes menu commands' `Enabled` state by invoking the associated command's `CanExecute` method, and so the idea is to log the exception so there's a trace of something going wrong if anything *looks* wrong in the VBE, but we don't fail initialization altogether just because some bug in some `CanExecute` method threw a NRE.

Would it make sense to `catch (COMException exception) { _logger.Error(exception); throw; }` then?

Reply:
Any exception bubbling out of the `Initialize()` scope can derail the initialization of Rubberduck: this is *healthy* over-thinking.

Reply:
I was thinking that the initialization may need to get separated into two parts, one that consists of operations that cannot fail silently and then the other that we don’t care whether it fails.  However that would add considerably more complexity for small gains, I think catching COM exceptions from VBIDE makes for good candidates for rethrowing and therefore failing the entire startup. But, can menu items themselves throw a COM exception?